build/compile.sh

```diff
@@ -171,22 +171,26 @@ build() {
 
   pushd $PY27
 
-  local compile_readline
-  local link_readline
-  if test "$HAVE_READLINE" = 1; then
+  local readline_flags
+  if [[ "$HAVE_READLINE" -eq 1 ]]; then
     # Readline interface for tokenizer.c and [raw_]input() in bltinmodule.c.
     # For now, we are using raw_input() for the REPL.  TODO: Parameterize this!
     # We should create a special no_readline_raw_input().
 
     c_module_src_list=$(cat $abs_c_module_srcs)
+
     # NOTE: pyconfig.h has HAVE_LIBREADLINE but doesn't appear to use it?
-    compile_readline='-D HAVE_READLINE'
-    link_readline='-l readline'
+    readline_flags=(-D HAVE_READLINE)
+
+    if [[ -n "$READLINE_DIR" ]]; then
+      readline_flags+=(-L "$READLINE_DIR/lib" -I "$READLINE_DIR/include")
+    fi
+    readline_flags+=(-l readline)
```

Comment:
Can we put `-D HAVE_READLINE` here instead of at the beginning to slightly simplify the code?

I know that order matters for `-l` flags, but I think `-D` should be OK to move around.